internal/services/amlfilesystem/aml_file_system_resource_test.go

```diff
@@ -0,0 +1,471 @@
+}
+
+resource "azurerm_key_vault_key" "test" {
+  name         = "test"
+  key_vault_id = azurerm_key_vault.test.id
+  key_type     = "RSA"
+  key_size     = 2048
+  key_opts     = ["decrypt", "encrypt", "sign", "unwrapKey", "verify", "wrapKey"]
+
+  depends_on = [
+    azurerm_key_vault_access_policy.client,
+    azurerm_key_vault_access_policy.server,
+  ]
+}
+
+resource "azurerm_storage_account" "test" {
+  name                            = "acctestsa%s"
+  resource_group_name             = azurerm_resource_group.test.name
+  location                        = azurerm_resource_group.test.location
+  account_tier                    = "Standard"
+  account_replication_type        = "LRS"
+  allow_nested_items_to_be_public = true
+}
+
+resource "azurerm_storage_container" "test" {
+  name                  = "storagecontainer"
+  storage_account_name  = azurerm_storage_account.test.name
+  container_access_type = "private"
+}
+
+resource "azurerm_storage_container" "test2" {
+  name                  = "storagecontainer2"
+  storage_account_name  = azurerm_storage_account.test.name
+  container_access_type = "private"
+}
+
+data "azuread_service_principal" "test" {
+  display_name = "HPC Cache Resource Provider"
+}
+
+resource "azurerm_role_assignment" "test" {
+  scope                = azurerm_storage_account.test.id
+  role_definition_name = "Contributor"
+  principal_id         = data.azuread_service_principal.test.object_id
+}
+
+resource "azurerm_role_assignment" "test2" {
+  scope                = azurerm_storage_account.test.id
+  role_definition_name = "Storage Blob Data Contributor"
+  principal_id         = data.azuread_service_principal.test.object_id
+}
+
+resource "azurerm_aml_file_system" "test" {
+  name                   = "acctest-amlfs-%d"
+  resource_group_name    = azurerm_resource_group.test.name
+  location               = azurerm_resource_group.test.location
+  sku_name               = "AMLFS-Durable-Premium-250"
+  subnet_id              = azurerm_subnet.test.id
+  storage_capacity_in_tb = 8
+  zones                  = ["2"]
+
+  maintenance_window {
+    day_of_week        = "Friday"
+    time_of_day_in_utc = "22:00"
+  }
+
+  identity {
+    type = "UserAssigned"
+
+    identity_ids = [
+      azurerm_user_assigned_identity.test.id
+    ]
+  }
+
+  key_encryption_key {
+    key_url         = azurerm_key_vault_key.test.id
+    source_vault_id = azurerm_key_vault.test.id
+  }
+
+  hsm_setting {
+    container_id         = azurerm_storage_container.test.resource_manager_id
+    logging_container_id = azurerm_storage_container.test2.resource_manager_id
+    import_prefix        = "/"
+  }
+
+  tags = {
+    Env = "Test"
+  }
+
+  depends_on = [azurerm_role_assignment.test, azurerm_role_assignment.test2]
+}
+`, r.template(data), data.RandomInteger, data.RandomString, data.RandomString, data.RandomInteger)
+}
+
+func (r AMLFileSystemResource) update(data acceptance.TestData) string {
```

Comment:
Can we reuse some configurations between the `complete` and `update`?

Reply:
Updated